src/corona_hakab_model/generation/circles_consts.py

```diff
@@ -93,12 +93,34 @@ def get_age_distribution(self):
 
     def get_connection_types_prob_by_age(self):
         return {age: self.connection_type_prob_by_age_index[i] for i, age in enumerate(self.ages)}
-
+    
     # overriding hash and eq to allow caching while using un-hashable attributes
     __hash__ = object.__hash__
     __eq__ = object.__eq__
 
 
+    def get_required_adult_distributions(self):
+        students = 0
+        teachers = 0
+        for i in range(self.ages):
+            if self.ages[i] <= 18:
+                students += self.age_prob[i]*self.connection_type_prob_by_age_index[i][ConnectionTypes.School]
+            else:
+                teachers += self.age_prob[i]*self.connection_type_prob_by_age_index[i][ConnectionTypes.School]
+                
+        teacher_student_ratio = teachers * 1.0 / students
+        school_sizes = self.circle_size_distribution_by_connection_type[ConnectionTypes.School][0]
+        
+        family_sizes = self.circle_size_distribution_by_connection_type[ConnectionTypes.Family][0]
+        family_distributions = {size: randint(1,2) for size in family_sizes if size==1}
+        family_distributions.update({size: rv_discrete(1, 2, values=([1, 2], [0.2, 0.8])) for size in family_sizes if size==2})
+        family_distributions.update({size: rv_discrete(1, 3, values=([1, 2, 3], [0.1, 0.8, 0.1])) for size in family_sizes if size > 2})
```

Comment:
is it possible that 3 adults will be in the same family? 
and in general, are those numbers based on the research team numbers?

Reply:
Yes, in case some are living with a grandparent